packages/x-tree-view/src/internals/plugins/useTreeViewLabel/useTreeViewLabel.test.tsx

```diff
@@ -0,0 +1,233 @@
+import { expect } from 'chai';
+import { act, fireEvent } from '@mui/internal-test-utils';
+import { describeTreeView } from 'test/utils/tree-view/describeTreeView';
+import { UseTreeViewLabelSignature } from '@mui/x-tree-view/internals';
+
+describeTreeView<[UseTreeViewLabelSignature]>(
+  'useTreeViewLabel plugin',
+  ({ render, setup, treeViewComponentName }) => {
+    describe('interaction', () => {
+      describe('render labelInput when needed', () => {
+        it('should not render labelInput when double clicked if item is not editable', function test() {
+          // This test is not relevant for the TreeItem component or the SimpleTreeView.
+          if (!setup.includes('TreeItem2') || treeViewComponentName.startsWith('SimpleTreeView')) {
```

Comment:
```suggestion
          if (treeViewComponentName !== 'TreeItem2' || treeViewComponentName.startsWith('SimpleTreeView')) {
```

That way when we drop `TreeItem2`, we will have a TS error and we won't forget to remove this check :+1: 

Same for the other tests